pycaption/microdvd.py

```diff
@@ -0,0 +1,105 @@
+from copy import deepcopy
+
+from .base import (
+    BaseReader, BaseWriter, CaptionSet, CaptionList, Caption, CaptionNode, DEFAULT_LANGUAGE_CODE)
+from .exceptions import CaptionReadNoCaptions, InvalidInputError
+import re
+
+
+class MicroDVDReader(BaseReader):
+    def detect(self, content):
+        return re.match("{\d+}{\d+}", content) is not None
+
+    def read(self, content, lang=DEFAULT_LANGUAGE_CODE):
+        if type(content) != str:
+            raise InvalidInputError('The content is not a unicode string.')
+
+        lines = content.splitlines()
+        captions = CaptionList()
+        fps = 25.0
+        for line in lines:
+            m = re.match(r"{(\d+)}{(\d+)}(.*)", line)
+            if not m: break
```

Comment:
If a line is empty we should skip it and use `continue` but if its format is wrong, I think we should raise an exception rather than stopping the reading without any warnings.

Reply:
@ana-nichifor Please have a look. I also added a test.